### DISCUSSION TITLE
Proposal: For `Flow<T>` returning functions to not be 'suspending'

### DISCUSSION THREAD

**Discussion by chris-hatton** [Kotlin/kotlinx-rpc#232] [General]
Clearly, member-functions of RPC Services that return an immediate type need to be `suspending`, reflecting the asynchronous nature of the contract. e.g:
```kotlin
@Rpc
interface MyService: RpcService {
    suspend fun getValue(): String // Will be implemented by asynchronous operations
}
```

Where a member-function returns `Flow<T>`, however, I believe it would be idiomatic for these **not** to be suspending i.e:
```kotlin
@Rpc
interface MyService: RpcService {
    suspend fun getValue(): String
    suspend fun getValueFlow(): Flow<String> // Currently this needs to be marked `suspend`
}
```
...I propose should be:
```kotlin
@Rpc
interface MyService: RpcService {
    suspend fun getValue(): String
    fun getValueFlow(): Flow<String>
}
```
`Flow`s are already asynchronous by nature; they can be returned immediately, and only the first element is delayed as necessary.
Making these `suspend` seems redundant - unless I am missing a reason why this has to be?

**Comment by Mr3zee** [comment]
Hey, @chris-hatton !
We just had a design session last week that came to the same conclusion.
This will be changed is some near future

**Comment by Mr3zee** [comment]
Done in 0.6.0
